backend/cmd/headlamp.go

```diff
@@ -1289,6 +1290,149 @@ func (c *HeadlampConfig) deleteCluster(w http.ResponseWriter, r *http.Request) {
 	c.getConfig(w, r)
 }
 
+// Get path of kubeconfig from source.
+func (c *HeadlampConfig) getKubeConfigPath(source string) (string, error) {
+	if source == "kubeconfig" {
+		return c.kubeConfigPath, nil
+	}
+
+	return defaultKubeConfigPersistenceFile()
+}
+
+// Handler for renaming a cluster.
+//
+//nolint:funlen
```

Comment:
I think we can prevent this by moving the looping part of config.Contexts into a different function and letting it handle the contextName setting?? Would that be something difficult to achieve??? Does it add more complexity??

Reply:
It might not be easy because we have logging which should be part of k8s handler which creates additional lines. Even if we move some logic to other function it is not possible with clear code to have it under 60 with logging involved.

Reply:
I agree with Ashu and the linter about this function being too long. It will be easier to add tests then as well.